src/transformStateWithClones.js

```diff
@@ -1,13 +1,39 @@
 'use strict';
-
-/**
- * @param {Object} state
- * @param {Object[]} actions
- *
- * @return {Object[]}
- */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const current = { ...state };
+  const resultingArray = [];
+  let i = 0;
+
+  for (const key of actions) {
+    switch (key.type) {
+      case 'addProperties':
+        Object.assign(current, key.extraData);
+        break;
+
+      case 'removeProperties':
+
+        for (const property of key.keysToRemove) {
+          delete current[property];
+        }
+        break;
+
+      case 'clear':
+        for (const property in current) {
+          delete current[property];
+        }
+        break;
+
+      default:
+
+        return state;
+    }
+
+    resultingArray[i] = { ...current };
+    i++;
```

Comment:
try using `push` method, you can learn more [here](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/push)